### ISSUE TITLE
Notes on MacOS instructions

### ISSUE THREAD

**Issue by shriv** [b-rodrigues/rix#140]
Hi there, 
I'm a super n00b with `nix` though I have used R with `renv` and Docker a fair bit. I'm now trying to pick up nix for dev envs so  I followed the instructions to set up an R environment on my M1 Mac with the MacOS vignette. Thanks heaps for this package! I'm looking forward to following the other vignettes in due course. I've made some notes on following the MacOS instructions below. I'm not sure how helpful they are so feel free to just close the issue after reading it. :-) 

I ran: 
- The `rix::init()` command. **Note the vignette has a small typo with `profile_action` instead of `rprofile_action`.**
```
rix_init(
  project_path = "./rix_darwin",
  rprofile_action = "create_missing"
)
```
- The `rix::rix()` set up. 
```
rix(
r_ver = "4.2.1",
    r_pkgs = [c](https://rdrr.io/r/base/c.html)("dplyr", "ggplot2"),
    system_pkgs = NULL,
    git_pkgs = NULL,
    ide = "other",
    project_path = ".",
    overwrite = TRUE)
```
- I ran `nix_build()` inside of Rstudio. I initially ran `nix-build` in the terminal but there was no output for over 5 minutes so I killed it and ran in Rstudio.  This seemed to finish successfully. Some messages of install progress would be helpful here especially for a beginner. 
- I closed Rstudio and ran `nix-shell` in the project folder. This command didn't respond for a while and after a few minutes when I ran `Ctrl-C` it turned out the shell had in fact started. I'm not sure if the terminal needs to restarted after `nix-build`?

## General notes
- I use `renv` quite a lot and I have to admit, I find it difficult not seeing an explicit snapshot of the packages installed by nix. Is there a way I can generate this so that other users can have pinned package dependencies installed? I don't know how many versions of a given package a channel has. I assume it could have more than one..? If this is something that will be switched on when you transition to flakes, I'm happy to wait. I might just do an `renv::snapshot()` in the interim. 
- Given the lack of Rstudio via nixpkgs, I will need to learn VS Code + R. `langserver` is installed by `rix` if `code` is chosen as the IDE. I assume any extensions within VS Code just need to be handled manually? 

## System Info
- M1 Apple Silicon 
- Sonoma 14.2.1 (23C71)
- Nix 2.18.1
- R 4.2.1 on system.

**Comment by b-rodrigues** [comment]
Thanks, fixed the typo with: c3b3677bc63c340538e5d7208c07ff54b75ee5ed

>I ran nix_build() inside of Rstudio. I initially ran nix-build in the terminal but there was no output for over 5 minutes so I killed it and ran in Rstudio.

Did you restart terminals and Rstudio after having installed Nix? Maybe that was the issue.

>Some messages of install progress would be helpful here especially for a beginner.

If you run `nix_build()` with `exec_mode = "non-blocking"` you'll have progress messages.

>I'm not sure if the terminal needs to restarted after nix-build?

No you shouldn't need to do so. But the first time you run `nix-shell` several more packages get downloaded, so it can take some time. Did you try running `which R` inside the nix shell? You should see a path to the Nix store.

> Is there a way I can generate this so that other users can have pinned package dependencies installed? I don't know how many versions of a given package a channel has. I assume it could have more than one..?

Pinning for reproducibility is done using the revision of `nixpkgs`. If you check your generated `default.nix` you'll see the revision that is used. This revision will always install exactly the same packages, and no  there's not more than one version of R package for one revision of `nixpkgs`. But indeed, running `renv::snapshot()` might be interesting if you want to list package versions.

>If this is something that will be switched on when you transition to flakes

This is not planned. I'm note sure there would be any benefit to using flakes for rix. But it's not a door I completely closed yet.

>I assume any extensions within VS Code just need to be handled manually?

Yes, you should be able to install VS Code using the macOS installer and configure it to your liking. Then, starting VS Code from the activated nix shell should make VS Code "see" the R environment installed from Nix.

You could also use `with_nix()` to run arbitrary code inside a subshell from your currently installed version of R from RStudio as well, see: https://b-rodrigues.github.io/rix/articles/z-advanced-topic-running-r-or-shell-code-in-nix-from-r.html

**Comment by shriv** [comment]
Thanks heaps for your comments @b-rodrigues! I forgot to mention that I had installed nix a couple weeks ago and had since restarted the terminal since install. I hadn't realised that `nix-shell` does some installation too so it makes sense that there was a wait till the shell started 👍 

I will follow the other vignettes in due course :-)
